torch/utils/data/datapipes/iter/combinatorics.py

```diff
@@ -94,65 +98,82 @@ def __init__(self,
                  unbatch_level: int = 0
```

Comment:
Should we accept `seed` as an optional argument here? What's downside if we do that?
It looks like the intention here for the users to pass RNG from DataLoader rather than directly setting one?

Reply:
Since users want to run shuffle with different order per epoch, providing a single seed is not enough.
You can treat the RNG in DataLoader as the single source of SeedGenerator. All DataPipe with random op should fetch seeds from it.
It's easier to maintain.